fineract-provider/src/main/java/org/apache/fineract/portfolio/paymenttype/service/PaymentTypeReadPlatformServiceImpl.java

```diff
@@ -41,6 +42,7 @@ public PaymentTypeReadPlatformServiceImpl(final PlatformSecurityContext context,
     }
 
     @Override
+    @Cacheable(value = "payment_types", key = "T(org.apache.fineract.infrastructure.core.service.ThreadLocalContextUtil).getTenant().getTenantIdentifier().concat('payment_types')")
```

Comment:
Should enhance the PaymentTypeWriteServiceImpl to invalidate the cache entries

Reply:
Good point, now the cahce is getting evicted.